### PR TITLE
[2.13.x] DDF-4018 closing input streams in catalogServiceImpl

### PR DESCRIPTION
**Pull request by Kjames5269**
#### What does this PR do?
Closes an input stream in CatalogServiceImpl

#### Who is reviewing it? 
@bakejeyner 
@blen-desta 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@coyotesqrl
@stustison 
#### How should this be tested?
1) Install DDF
2) Import a file into DDF hitting `https://localhost:8993/services/catalog/metacard`
  Nothing should break 👍 

#### What are the relevant tickets?
[DDF-4018](https://codice.atlassian.net/browse/DDF-4018)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Kjames5269** [comment]
build now

File: catalog/rest/catalog-rest-impl/src/main/java/org/codice/ddf/rest/impl/CatalogServiceImpl.java [hunk 1]
```diff
@@ -548,6 +548,14 @@ private BinaryContent createMetacard(
 
     } catch (MetacardCreationException | CatalogTransformerException e) {
       throw new CatalogServiceException("Unable to create metacard");
+    } finally {
+      try {
+        if (stream != null) {
+          stream.close();
+        }
+      } catch (IOException e) {
+        LOGGER.error("Unexpected error closing stream", e);
```
Comment (rzwiefel) [inline catalog/rest/catalog-rest-impl/src/main/java/org/codice/ddf/rest/impl/CatalogServiceImpl.java]: ❓ Do we want these at error? 
https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels

**Comment by blen-desta** [comment]
Hero: Successfully created a metacard by hitting https://localhost:8993/services/catalog/metacard

**Comment by coyotesqrl** [comment]
build now

**Comment by Kjames5269** [comment]
build now
(aborted and can't see the logs)

**Pull request merged**